packages/block-editor/src/hooks/border.js

```diff
@@ -44,9 +50,37 @@ export function hasBorderSupport( blockName ) {
 
 	const support = getBlockSupport( blockName, BORDER_SUPPORT_KEY );
 
-	// Further border properties to be added in future iterations.
-	// e.g. support && ( support.radius || support.width || support.style )
-	return !! ( true === support || support?.radius );
+	return !! (
+		true === support ||
+		support?.color ||
+		support?.radius ||
+		support?.width ||
+		support?.style
+	);
+}
+
+/**
+ * Determines if there a specific border feature is supported.
+ *
+ * @param {string}        feature   Name of the border feature e.g.`radius`
+ * @param {string|Object} blockType Block name or block type object.
+ * @return {boolean}                Whether the border feature is supported.
+ */
+export function hasBorderFeatureSupport( feature, blockType ) {
```

Comment:
Can we merge `hasBorderSupport` and `hasBorderFeatureSupport` into a single function `hasBorderSupport( blockType, feature = 'all' )` or something along those lines?

Reply:
Related https://github.com/WordPress/gutenberg/pull/30124/files?file-filters%5B%5D=.js#r612670345

Reply:
This has been refactored ✅ 

Thanks.